api/base_controller.go

```diff
@@ -260,6 +268,30 @@ func (c *BaseController) DeleteSingleObject(r *web.Request) (*web.Response, erro
 	if err != nil {
 		return nil, fmt.Errorf("could not generate GUID for %s: %s", c.objectType, err)
 	}
+	var cascadeRootId = ""
+	if opCtx.Cascade {
+		if c.supportsCascadeDelete {
+			// Scan if requested resource really exists
+			resources, err := c.repository.List(ctx, c.objectType, criteria...)
```

Comment:
why do we need to check this here? is it different than non-cascading flow? who's the one to return 404 when it is not cascade?

Reply:
If not to do a check here, then in case when resource is not found we are getting "/v1/operation not found" error instead of. As the message is confusing, we have a check in advance to provide more accurate error message.